rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/RxBleConnectionImpl.java

```diff
@@ -195,6 +204,10 @@ public int getMtu() {
     @Override
     public Observable<Observable<byte[]>> setupIndication(@NonNull BluetoothGattCharacteristic characteristic,
                                                           @NonNull NotificationSetupMode setupMode) {
+        illegalOperationChecker.checkAnyPropertyMatches(
```

Comment:
I agree that there should be a model separation. The method should check whether preconditions are correct and the result should be described by the caller.